pkg/reconciler/autoscaling/hpa/resources/keda.go

```diff
@@ -184,3 +236,68 @@ func getDefaultPrometheusTrigger(annotations map[string]string, address string,
 	trigger.AuthenticationRef = ref
 	return &trigger, nil
 }
+
+func getExtraPrometheusTriggers(annotations map[string]string) ([]v1alpha1.ScaleTriggers, error) {
+	triggers := []v1alpha1.ScaleTriggers(nil)
+	if v, ok := annotations[KedaAutoscaleAnnotationExtraPrometheusTriggers]; ok {
+		if err := json.Unmarshal([]byte(v), &triggers); err != nil {
+			return nil, fmt.Errorf("unable to unmarshal extra prometheus triggers: %w", err)
+		}
+	}
+	return triggers, nil
+}
+
+func getMetricType(annotations map[string]string, metric string) (*autoscalingv2.MetricTargetType, error) {
+	var mt autoscalingv2.MetricTargetType
+	v, ok := annotations[KedaAutoscaleAnnotationMetricType]
+	if ok {
+		if v != string(autoscalingv2.AverageValueMetricType) && v != string(autoscalingv2.UtilizationMetricType) && v != string(autoscalingv2.ValueMetricType) {
+			return nil, fmt.Errorf("invalid metric type: %s", v)
+		}
+	}
+	switch metric {
+	case autoscaling.CPU:
+		if ok {
+			if v == string(autoscalingv2.ValueMetricType) {
```

Comment:
this seems to be repeated/duplicated with the above (l 254). Could the two checks be combined in the switch?